src/lib/src/cube/cube.cpp

```diff
@@ -0,0 +1,183 @@
+
+namespace py = pybind11;
+
+namespace xtgeo::cube {
+
+/*
+ * Compute cube statistics per column, returning 2D arrays of min, max, mean, var, rms,
+ * maxabs, maxpos, maxneg, meanabs, meanpos, meanneg, sumpos, sumneg, sumabs. This could
+ * technically be done in numpy, but this approach is more efficient as it can do all
+ * operations per column in one operation.
+ *
+ * @param ncol Cube dimensions ncol
+ * @param nrow Cube dimensions nrow
+ * @param nlay Cube dimensions nlay
+ * @param cubev Cube values vector
+ * @return 14 arrays: min, max, mean, var, rms, maxpos, maxneg, maxabs,
+ * meanpos, meanneg, meanabs, sumpos, sumneg, sumabs
+ */
+
+std::tuple<py::array_t<double>,
+           py::array_t<double>,
+           py::array_t<double>,
+           py::array_t<double>,
+           py::array_t<double>,
+           py::array_t<double>,
+           py::array_t<double>,
+           py::array_t<double>,
+           py::array_t<double>,
+           py::array_t<double>,
+           py::array_t<double>,
+           py::array_t<double>,
+           py::array_t<double>,
+           py::array_t<double>>
+cube_stats_along_z(const size_t ncol,
+                   const size_t nrow,
+                   const size_t nlay,
+                   const py::array_t<float> &cubev)
+{
+    py::array_t<double> minv({ ncol, nrow });
+    py::array_t<double> maxv({ ncol, nrow });
+    py::array_t<double> meanv({ ncol, nrow });
+    py::array_t<double> varv({ ncol, nrow });
+    py::array_t<double> rmsv({ ncol, nrow });
+    py::array_t<double> maxposv({ ncol, nrow });
+    py::array_t<double> maxnegv({ ncol, nrow });
+    py::array_t<double> maxabsv({ ncol, nrow });
+    py::array_t<double> meanposv({ ncol, nrow });
+    py::array_t<double> meannegv({ ncol, nrow });
+    py::array_t<double> meanabsv({ ncol, nrow });
+    py::array_t<double> sumposv({ ncol, nrow });
+    py::array_t<double> sumnegv({ ncol, nrow });
+    py::array_t<double> sumabsv({ ncol, nrow });
+
+    auto cubev_ = cubev.unchecked<3>();  // Use unchecked for efficiency
+    auto minv_ = minv.mutable_unchecked<2>();
+    auto maxv_ = maxv.mutable_unchecked<2>();
+    auto meanv_ = meanv.mutable_unchecked<2>();
+    auto varv_ = varv.mutable_unchecked<2>();
+    auto rmsv_ = rmsv.mutable_unchecked<2>();
+    auto maxposv_ = maxposv.mutable_unchecked<2>();
+    auto maxnegv_ = maxnegv.mutable_unchecked<2>();
+    auto maxabsv_ = maxabsv.mutable_unchecked<2>();
+    auto meanposv_ = meanposv.mutable_unchecked<2>();
+    auto meannegv_ = meannegv.mutable_unchecked<2>();
+    auto meanabsv_ = meanabsv.mutable_unchecked<2>();
+    auto sumposv_ = sumposv.mutable_unchecked<2>();
+    auto sumnegv_ = sumnegv.mutable_unchecked<2>();
+    auto sumabsv_ = sumabsv.mutable_unchecked<2>();
+
+    for (size_t i = 0; i < ncol; i++) {
+        for (size_t j = 0; j < nrow; j++) {
+            double min_val = std::numeric_limits<double>::max();
+            double max_val = std::numeric_limits<double>::lowest();
```

Comment:
Are these correct? `min_val = ::max()`, `max_val = ::lowest()`

Reply:
I think they are correct: initialize min to a very large number and max to a very large negative number (hence ::lowest(), not ::min() )